xapian-core/tests/api_weight.cc

```diff
@@ -1625,3 +1625,63 @@ DEFINE_TESTCASE(coordweight2, !backend) {
     }
     return true;
 }
+
+// Test exception for junk after serialised weight.
+DEFINE_TESTCASE(dicecoeffweight1, !backend) {
+    Xapian::DiceCoeffWeight wt;
+    try {
+	Xapian::DiceCoeffWeight t;
+	Xapian::DiceCoeffWeight * t2 = t.unserialise(wt.serialise() + "X");
+	// Make sure we actually use the weight.
+	bool empty = t2->name().empty();
+	delete t2;
+	if (empty)
+	    FAIL_TEST("Serialised DiceCoeffWeight with junk appended\
+		    unserialised to empty name!");
```

Comment:
Don't carry string constants across a newline line this - you'll end up with a load of unwanted spaces in the middle of the string.  Instead break the string into a string constant on each line - they'll get combined into a single string constant by the compiler.

Reply:
Demo of problem with your code:

```
$ cat > hellow.cc
#include <iostream>
int main() {
std::cout << "hello \
              world\n";
}
$ g++ -Wall -W -O2 hellow.cc
$ ./a.out 
hello               world
```

Reply:
I see! Thanks for the demo. Fixed now.